bot/exts/info/tags.py

```diff
@@ -24,145 +29,214 @@
-        return found
-
-    def _get_tags_via_content(self, check: Callable[[Iterable], bool], keywords: str, user: Member) -> list:
+    def get_fuzzy_matches(self, tag_identifier: TagIdentifier) -> list[tuple[TagIdentifier, Tag]]:
+        """Get tags with identifiers similar to `tag_identifier`."""
+        if tag_identifier.group is None:
+            if len(tag_identifier.name) < 3:
+                return []
+            else:
+                return self._get_suggestions(tag_identifier)
+        else:
+            if len(tag_identifier.group) < 3:
+                suggestions = []
+            else:
+                # Try fuzzy matching with only a name first
+                suggestions = self._get_suggestions(TagIdentifier(None, tag_identifier.group))
+
+            if not len(tag_identifier.name) < 3:
+                suggestions += self._get_suggestions(tag_identifier)
+            return suggestions
+
+    def _get_tags_via_content(
+            self,
+            check: Callable[[Iterable], bool],
+            keywords: str,
+            user: Member,
+    ) -> list[tuple[TagIdentifier, Tag]]:
         """
         Search for tags via contents.
 
         `predicate` will be the built-in any, all, or a custom callable. Must return a bool.
         """
-        keywords_processed: List[str] = []
-        for keyword in keywords.split(','):
+        keywords_processed = []
+        for keyword in keywords.split(","):
             keyword_sanitized = keyword.strip().casefold()
             if not keyword_sanitized:
                 # this happens when there are leading / trailing / consecutive comma.
                 continue
             keywords_processed.append(keyword_sanitized)
 
         if not keywords_processed:
-            # after sanitizing, we can end up with an empty list, for example when keywords is ','
+            # after sanitizing, we can end up with an empty list, for example when keywords is ","
             # in that case, we simply want to search for such keywords directly instead.
             keywords_processed = [keywords]
 
         matching_tags = []
-        for tag in self._cache.values():
-            matches = (query in tag['embed']['description'].casefold() for query in keywords_processed)
-            if self.check_accessibility(user, tag) and check(matches):
-                matching_tags.append(tag)
+        for identifier, tag in self._tags.items():
+            matches = (query in tag.content.casefold() for query in keywords_processed)
+            if tag.accessible_by(user) and check(matches):
+                matching_tags.append((identifier, tag))
 
         return matching_tags
 
-    async def _send_matching_tags(self, ctx: Context, keywords: str, matching_tags: list) -> None:
+    async def _send_matching_tags(
+            self,
+            ctx: Context,
+            keywords: str,
+            matching_tags: list[tuple[TagIdentifier, Tag]],
```

Comment:
We define this type a lot, isn't there a way to create an alias. As in:
```py
tag_pair = tuple[TagIdentifier, Tag]
```
Then this code can become:
```suggestion
            matching_tags: list[tag_pair],
```
May not be a huge benefit, but I thought I'd mention it.

Reply:
Although type aliases should be in PascalCase

Reply:
Yes, that'd be nicer, although I don't think `TagPair` is a fitting name as I'd expect a pair of `Tag` objs from that, not sure on a good name myself